addons/website_slides/controllers/slide.py

```diff
@@ -0,0 +1,82 @@
+from odoo.addons.http_routing.models.ir_http import slug
+from odoo.addons.website.models.ir_http import sitemap_qs2dom
+
+
+_logger = logging.getLogger(__name__)
+
+class SlideController(http.Controller):
+  
+
+    @http.route('/get/quiz', type="json", auth="public", website=True)
+    def get_quiz(self, **kw):
+        if 'slide_id' in kw:
+            slide = request.env['slide.slide'].browse(kw['slide_id'])
+            slide_user = request.env['slide.slide.partner'].search([('slide_id', '=', slide.id), ('partner_id', '=', request.env.user.partner_id.id)])
+            possible_points = ["quiz_first_attempt_reward","quiz_second_attempt_reward","quiz_third_attempt_reward"]
+            res = {
+                'questions':[{'title': question.question,'id': question.id,'answers': [{'text': answer.text_value, 'correct':answer.is_correct,'id': answer.id} for answer in question.answer_ids]} for question in slide.question_ids ],
+                'nb_attempts': slide_user.quiz_attempts_count if slide_user else 0,
+                'possible_rewards': [slide.quiz_first_attempt_reward, slide.quiz_second_attempt_reward, slide.quiz_third_attempt_reward, slide.quiz_fourth_attempt_reward],
+                'reward': (slide[possible_points[slide_user.quiz_attempts_count]] if slide_user.quiz_attempts_count < 3 else slide.quiz_fourth_attempt_reward) if slide_user else slide.quiz_first_attempt_reward
+            }
+            return res
+
+    @http.route('/slide/quiz/submit', type="json", auth="user", website=True)
+    def submit_quiz(self, slide_id, answer_ids,**kw):
+        slide = request.env['slide.slide'].browse(slide_id)
+        answers = request.env['slide.answer'].search([('id', 'in', answer_ids)])
+        good_answers = answers.search([('id', 'in', answer_ids),('is_correct', '=', True)])
+        bad_answers = answers - good_answers
+        slide_partner = request.env['slide.slide.partner'].search([('slide_id', '=', slide_id),('partner_id', '=', request.env.user.partner_id.id)])
+        possible_points = ["quiz_first_attempt_reward","quiz_second_attempt_reward","quiz_third_attempt_reward"]
+        points = 0
+        if not slide_partner:
+            slide_partner = request.env['slide.slide.partner'].sudo().create({
+                "slide_id": slide.id,
+                "partner_id": request.env.user.partner_id.id,
+                "channel_id": slide.channel_id.id,
+                "completed": False
+            })
+        if not slide_partner.completed:
+            if slide_partner.quiz_attempts_count < 3:
+                points = slide[possible_points[slide_partner.quiz_attempts_count]]
+            else:
+                points = slide.quiz_fourth_attempt_reward
+            slide_partner.sudo().write({
+                'quiz_attempts_count': slide_partner.quiz_attempts_count if not bad_answers else slide_partner.quiz_attempts_count + 1,
+                'points_won': points if not bad_answers else 0,
```

Comment:
If you have a single bad answer you have 0 points_won ? Might be the case but strange.

Reply:
Yup, a quiz will only be considered as complete when the user answers everything right. Therefore, 0 points as long as he makes mistakes. Once he got all answers right, a set amount points will be given to him depending on the number of attempts he made.